grading/load_config_json.cpp

```diff
@@ -123,37 +123,58 @@ void AddGlobalDefaults(nlohmann::json &whole_config) {
     whole_config["gradeable_message"] = whole_config.value("assignment_message","");
```

Comment:
are these changes meant for the arm work? Is this because getting a newer version of the `nlohmann::json` library?

Reply:
Yes, I hate these changes.
I think it's a different version of nlohmann::json.  
Or perhaps a different version of the compiler, or both.

Reply:
In PR #6833, I've got us properly using the most recent version of the the nlohmann json library.  And I think it's using the same version of compiler on both x86 and arm machines, but unfortunately I'm getting null/type errors with the original version of the code only on arm.  Neither version of the code is elegant -- lots of this should be refactor eventually.